tools/walletextension/accountmanager/account_manager.go

```diff
@@ -59,20 +61,82 @@ func (m *AccountManager) AddClient(address gethcommon.Address, client *rpc.EncRP
 // ProxyRequest tries to identify the correct EncRPCClient to proxy the request to the Obscuro node, or it will attempt
 // the request with all clients until it succeeds
 func (m *AccountManager) ProxyRequest(rpcReq *RPCRequest, rpcResp *interface{}, userConn userconn.UserConn) error {
+	if rpcReq.Method == rpc.Subscribe {
+		client, err := m.suggestSubscriptionClient(rpcReq)
+		if err != nil {
+			return err
+		}
+		// fetch the client from a topic
+		return m.executeSubscribe(client, rpcReq, rpcResp, userConn)
+	}
+
+	return m.executeCall(rpcReq, rpcResp)
+}
+
+const emptyFilterCriteria = "[]" // This is the value that gets passed for an empty filter criteria.
+
+func (m *AccountManager) suggestSubscriptionClient(rpcReq *RPCRequest) (rpc.Client, error) {
+	var client rpc.Client //= m.unauthedClient // todo - support lifecycle events through an unauthed client
+
+	// by default use the first client for lifecycle events
+	for _, c := range m.accountClients {
+		client = c
+		break
+	}
+
+	// determine the client based on the topics
+	// if none is found use the unauthed client and assume this is a lifecycle method
+	if len(rpcReq.Params) == 0 {
+		return client, nil
+	}
+
+	filterCriteriaJSON, err := json.Marshal(rpcReq.Params[1])
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal filter criteria to JSON. Cause: %w", err)
+	}
+
+	filterCriteria := filters.FilterCriteria{}
+	if string(filterCriteriaJSON) != emptyFilterCriteria {
+		err = filterCriteria.UnmarshalJSON(filterCriteriaJSON)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal filter criteria from the following JSON: `%s`. Cause: %w", string(filterCriteriaJSON), err)
+		}
+	}
+	for i, topicCondition := range filterCriteria.Topics {
+		if i == 0 {
+			continue
+		}
+		for _, topic := range topicCondition {
+			bitlen := topic.Big().BitLen()
+			if bitlen < 80 || bitlen > 160 {
```

Comment:
This is all a bit magic, probably worth a comment to explain what's being checked here. Maybe even worth pulling out the details of this check into a `doesTopicContainClientAddress()` method or something for readability.